elastalert/elastalert.py

```diff
@@ -1591,7 +1592,8 @@ def send_alert(self, matches, rule, alert_time=None, retried=False):
                 matches[0]['kibana_link'] = kb_link
 
         if rule.get('generate_kibana_discover_url'):
-            kb_link = generate_kibana_discover_url(rule, matches[0])
+            shortener = pyshorteners.Shortener()
+            kb_link = shortener.tinyurl.short(generate_kibana_discover_url(rule, matches[0]))
```

Comment:
This should be optional.  I personally don't want an external dependency on

https://tinyurl.com

Kibana has built in short link functionality.  Might be better to tie to that.

Reply:
>Kibana has built in short link functionality. Might be better to tie to that.

How do you achieve it? .. Is there an implementation example? Or will you do it? ..

>This should be optional. I personally don't want an external dependency on

Did you read the question?
https://github.com/jertel/elastalert2/discussions/445

Reply:
Api for shortening

https://www.elastic.co/guide/en/kibana/master//url-shortening-api.html

Reply:
Does generate_kibana_discover_short_url mean more settings?

Reply:
Likely.  Short urls are documents in Kibana's index, so not everyone is going to want to come up with an expiration strategy right away.

Reply:
>https://www.elastic.co/guide/en/kibana/master//url-shortening-api.html

You may be curious about the Shorten URL API. Since kibana may have another server, do you expect to add kibana_shorten_api_url as well?
Also, is there a case where user / password is required to execute the Shorten URL API?

Reply:
I found the [Python library for kibana-api](https://github.com/cr0wg4n/kibana-api). Looking at the implementation, there seems to be a pattern that requires user and password. In that case, the following settings will be added.

- kibana_shorten_api_url  (string)
- kibana_username (string)
- kibana_password (string)
- generate_kibana_discover_short_url (boolean。default : False)

Reply:
So in the case of opendistro it could actually be AWS signed as well.  All the same authentication methods would apply.